rpcs3/Emu/Cell/Modules/sceNp.cpp

```diff
@@ -2335,24 +2481,28 @@ error_code sceNpManagerGetOnlineId(vm::ptr<SceNpOnlineId> onlineId)
 		return SCE_NP_ERROR_INVALID_ARGUMENT;
 	}
 
-	if (g_psn_connection_status == SCE_NP_MANAGER_STATUS_OFFLINE)
+	if (nph->get_psn_status() == SCE_NP_MANAGER_STATUS_OFFLINE)
 	{
 		return not_an_error(SCE_NP_ERROR_OFFLINE);
 	}
 
-	if (g_psn_connection_status != SCE_NP_MANAGER_STATUS_LOGGING_IN && g_psn_connection_status != SCE_NP_MANAGER_STATUS_ONLINE)
+	if (nph->get_psn_status() != SCE_NP_MANAGER_STATUS_LOGGING_IN && nph->get_psn_status() != SCE_NP_MANAGER_STATUS_ONLINE)
 	{
 		return SCE_NP_ERROR_INVALID_STATE;
 	}
 
+	memcpy(onlineId.get_ptr(), &nph->get_online_id(), sizeof(SceNpOnlineId));
```

Comment:
Nitpick, but `sizeof(nph->get_online_id())` may work. Or something similar, to tie the size to this function's return type.

Reply:
yeah but it'd call the function twice and return type is not going to change, keeping as is.